lib/system_description_validator.rb

```diff
@@ -16,195 +16,75 @@
-    message
-  end
-
   private
 
-  def load_global_schema(format_version = SystemDescription::CURRENT_FORMAT_VERSION)
-    JSON.parse(File.read(File.join(
-      Machinery::ROOT,
-      "schema",
-      "v#{format_version}",
-      "system-description-global.schema.json"
-    )))
-  end
-
-  def load_scope_schemas(format_version = SystemDescription::CURRENT_FORMAT_VERSION)
-    schema_dir = File.join(
-      Machinery::ROOT,
-      "plugins",
-      "schema",
-      "v#{format_version}",
-    )
-
-    Hash[
-      Dir["#{schema_dir}/*.schema.json"].map do |file|
-        scope = file.match(/system-description-(.*)\.schema\.json$/)[1].tr("-", "_")
-        schema = JSON.parse(File.read(file))
-
-        [scope, schema]
-      end
-    ]
-  end
-
-  def cleanup_json_path(message, scope)
-    old_path = message[/The property '#\/(.*?)'/,1]
-
-    position = error_position_from_json_path(old_path)
-    details = extract_details_from_json_path(old_path, scope)
-
-    new_path = "The property"
-    new_path += " ##{position}" if position > -1
-    new_path += " (#{details})" if !details.empty?
-
-    message.gsub(/The property '#\/.*?'/, new_path)
-  end
-
-  def error_position_from_json_path(path)
-    elements = path.split("/")
-
-    position = -1
-    elements.each do |e|
-      if Machinery::is_int?(e)
-        number = e.to_i
-        position = number if number > position
-      end
+  def scope_extracted?(scope)
+    if @format_version == 1
+      @hash[scope] && ScopeFileStore.new(@path, scope.to_s).path
+    else
+      @hash[scope] && @hash[scope]["extracted"]
     end
-    position
   end
 
-  def extract_details_from_json_path(path, scope)
-    elements = path.split("/")
-
-    elements.uniq!
-
-    # filter numbers since the position is calculated elswhere
-    elements.reject! { |e| Machinery::is_int?(e) }
-
-    # The json schema path often contains the word "type" in many messages
-    # but this information adds no value for the user since it is not related
-    # to our manifest.json
-    # So we filter it for all scopes except of repositories since this is the
-    # only scope which does have an attribute called "type"
-    if scope != "repositories"
-      elements.reject! { |e| e == "type" }
+  def expected_files(scope)
+    if @format_version == 1
+      files = @hash[scope]
+    else
+      files = @hash[scope]["files"]
     end
 
-    elements.join("/")
-  end
+    if scope == "unmanaged_files"
+      has_files_tarball = files.any? { |f| f["type"] == "file" || f["type"] == "link" }
+      tree_tarballs = files.
+        select { |f| f["type"] == "dir" }.
+        map { |d| File.join("trees", d["name"].sub(/\/$/, "") + ".tgz") }
+
+      expected_files = []
+      expected_files << "files.tgz" if has_files_tarball
+      expected_files += tree_tarballs
+    else
+      expected_files = files
+        .reject { |file| file["changes"].include?("deleted") }
+        .map {|file| file["name"] }
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Space between { and | missing.